src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private int defaultSize = 10;
+    private int maxSize = 50;
+    private int maxBound = 5;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+    int index = random.nextInt(maxBound);
+
+    private int randomSize() {
+        return 1 + random.nextInt(maxSize);
+    }
+
+    public Figure getRandomFigure() {
+        switch (index) {
+            case 0:
+                return new Circle(randomSize(), colorSupplier.getRandomColor());
+            case 1:
+                return new Square(randomSize(), colorSupplier.getRandomColor());
+            case 2:
+                return new Rectangle(colorSupplier.getRandomColor(), randomSize(), randomSize());
```

Comment:
Instead of passing expressions directly to the constructor, create separate variables for the sizes and pass them on for better code readability.